drainer/syncer.go

```diff
@@ -203,9 +206,10 @@ func (s *Syncer) handleSuccess(fakeBinlog chan *pb.Binlog, lastTS *int64) {
 				atomic.StoreInt64(lastTS, ts)
 			}
 
-			// save ASAP for DDL
-			if item.Binlog.DdlJobId > 0 {
+			// save ASAP for DDL, and if FinishTS > 0, we should save the ts map
+			if item.Binlog.DdlJobId > 0 || item.FinishTS > 0 {
 				saveNow = true
+				finishTS = item.FinishTS
```

Comment:
where to reset `finishTS`

Reply:
updated https://github.com/pingcap/tidb-binlog/pull/670/commits/4c8782f0e021d79d1b1ecb9094716dc0a9810c2f